cdap-notifications-api/src/main/java/co/cask/cdap/notifications/feeds/NotificationFeedManager.java

```diff
@@ -31,31 +33,32 @@
    * @throws NotificationFeedException if the feed has an incorrect structure;
    * for example, if it is missing a name, a namespace or a category
    */
-  boolean createFeed(NotificationFeed feed) throws NotificationFeedException;
+  boolean createFeed(Id.NotificationFeed feed) throws NotificationFeedException;
 
   /**
    * Deletes the {@code feed} from the manager store. To determine if the feed exists,
-   * the {@link NotificationFeed#getId} method is used.
+   * the {@link Id.NotificationFeed#getId} method is used.
    *
-   * @param feed the {@link NotificationFeed} to delete
+   * @param feed the {@link Id.NotificationFeed} to delete
    * @throws NotificationFeedNotFoundException if the feed does not exist
    */
-  void deleteFeed(NotificationFeed feed) throws NotificationFeedException;
+  void deleteFeed(Id.NotificationFeed feed) throws NotificationFeedException;
```

Comment:
Should this also throw a ``NotFoundException``.

Maybe unrelated to this PR, but something to think about.